crates/criticalup-core/src/project_manifest/v1.rs

```diff
@@ -1,21 +1,47 @@
 // SPDX-FileCopyrightText: The Ferrocene Developers
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use serde::Deserialize;
+use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
-#[derive(Deserialize)]
+#[derive(Deserialize, Serialize)]
 #[serde(deny_unknown_fields, rename_all = "kebab-case")]
-pub(super) struct ProjectManifest {
+pub struct ProjectManifest {
     #[allow(unused)]
     manifest_version: u32,
     #[serde(default)]
     pub(super) products: HashMap<String, ProjectManifestProduct>,
 }
 
-#[derive(Deserialize)]
+#[derive(Deserialize, Serialize)]
 #[serde(deny_unknown_fields, rename_all = "kebab-case")]
 pub(super) struct ProjectManifestProduct {
     pub(super) release: String,
     pub(super) packages: Vec<String>,
 }
+
+pub fn sample_manifest(release: Option<String>) -> ProjectManifest {
+    let packages = vec![
+        "cargo-${rustc-host}".into(),
+        "rustc-${rustc-host}".into(),
+        "clippy-${rustc-host}".into(),
+        "rust-std-${rustc-host}".into(),
+        "rustfmt-${rustc-host}".into(),
+    ];
+
+    let mut product = ProjectManifestProduct {
+        release: "".into(),
+        packages,
+    };
+
+    if let Some(release_val) = release {
+        product.release = release_val;
+    } else {
+        product.release = "stable-24.11.0".to_string();
```

Comment:
I was not sure what to do with the "default" release version. In this PR, I do give users a choice to pass the version via CLI but if nothing is passed, I chose "stable-24.11.0" for now.

Reply:
This circles back to some of our discussions about having the download-server able to present users with the latest stable via API etc. I wonder if it's worth doing that as part of this...

Reply:
Hrm, maybe `--release` should actually be a required arg...

Reply:
I think thats not entirely a bad idea. I was holding off but I just made it mandatory.